cluster/fsm/snapshot.go

```diff
@@ -0,0 +1,36 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package fsm
+
+// Snapshot is the snapshot of the cluster FSMs (schema, rbac, etc)
+// it is used to restore the Snapshot to a previous state,
+// or to bring out-of-date followers up to a recent log index.
+type Snapshot struct {
```

Comment:
Not totally sure what the best way is, but we could either include a Version here or in each entry (eg RBAC would have its own version)